src/state/apis/limit-orders/limitOrderApi.ts

```diff
@@ -211,16 +174,48 @@ export const limitOrderApi = createApi({
         }
       },
     }),
-    cancelLimitOrders: build.mutation<boolean, { payload: OrderCancellation; chainId: ChainId }>({
-      queryFn: async ({ payload, chainId }) => {
+    cancelLimitOrders: build.mutation<
+      boolean,
+      {
+        accountId: AccountId
+        sellAssetId: AssetId
+        buyAssetId: AssetId
+        order: Order
+        wallet: HDWallet | null
+      }
+    >({
+      queryFn: async ({ accountId, order, wallet }, { getState }) => {
+        const state = getState() as ReduxState
         const config = getConfig()
         const baseUrl = config.REACT_APP_COWSWAP_BASE_URL
-        const maybeNetwork = getCowswapNetwork(chainId)
-        if (maybeNetwork.isErr()) throw maybeNetwork.unwrapErr()
-        const network = maybeNetwork.unwrap()
+        const { chainId } = fromAccountId(accountId)
+        const accountMetadata = selectPortfolioAccountMetadataByAccountId(state, { accountId })
+
+        if (!wallet) throw Error('missing wallet')
+        if (!accountMetadata) throw Error('missing accountMetadata')
+
+        const signMessage = async (typedData: TypedData) => {
+          return await signCowMessage(
+            typedData,
+            assertGetEvmChainAdapter(chainId),
+            accountMetadata,
+            wallet,
+          )
+        }
+
+        const signature = await signCowOrderCancellation(order.uid, chainId, signMessage)
+
+        const network = assertGetCowNetwork(chainId)
+        const orderCancellationPayload: OrderCancellation = {
+          orderUids: [order.uid],
+          signature,
+          signingScheme: EcdsaSigningScheme.EIP712,
+        }
+
         const result = await axios.delete<void>(`${baseUrl}/${network}/api/v1/orders`, {
-          data: payload,
+          data: orderCancellationPayload,
         })
+
         // If the result is a 200 then the order was successfully canceled
         return { data: result.status === 200 }
```

Comment:
See https://github.com/shapeshift/web/pull/8203/files#r1867041490 - we probably want to add an early return for `result.status !== 200`, which I believe would translate in a try/catch and checking for `isAxiosError()` (iirc, axios defaults to throwing for all non-200s HTTP codes)

Reply:
What would the value be in each case though? We'd either have the current implemenation
```
{ data: isSuccessful }
```

or maybe something like
```
{ data: 'success'; error: 'failed' }
```

which seems redundant

Reply:
Essentially we need to detect an error case, so we require a value in each case to discriminate between them, so if `data` isnt a success flag, we'll have to somehow match on the value of `error` and `data`, so the value of each will matter.

Reply:
I may be wrong here since this is a mutation, not a query, but early returning `{error: true, status: axiosRes.status}` (or anything as `error` really`) is what I meant, so we could leverage the `error` property from the RTK mutation hook - point being that as it stands, this will just throw vs. `error` being returned explicitly, which would be cleaner to handle for consumers

Reply:
Ok so looking at types, this seems to already be automagically handled by mutations so no need to try/catch 🙏🏽 

<img width="1103" alt="image" src="https://github.com/user-attachments/assets/8c2fe078-beee-4ff8-999c-868ef63c8272">

Reply:
excellent, will do in follow-up!